### ISSUE TITLE
Drag delay with upgraded app

### ISSUE THREAD

**Issue by ckrahe** [nathancahill/Split.js#74]
We recently replaced our very old use of Split.js 0.4.3 with 1.2.0.  I read through the release notes, made sure to replace use of element id's with element id selectors, and don't recall any other significant changes we had to make.  All is working as expected but now there is a rather noticeable lag when dragging the split bars in our app.  Not just upon starting to drag, but at all times.  We have an onDragEnd callback, and that's firing as expected.  We do not register callbacks for either onDragStart or onDrag.  We also looked at a CSS2-style animation (transition) that we use to animate the panel as it opens for the first time, but shortening this or turning it off doesn't seem to resolve the drag lag.

Any thoughts on what might be causing this?

**Comment by nathancahill** [comment]
Thanks for the quality bug report. My goal is to make the upgrade process as painless as possible.

Indeed, replacing ids with selectors is the only change that affected most installs. Most of the other changes have been internal, but none of the changes should have altered performance, so I'm not sure why you're seeing lag now.

Do you see the same drag lag here? http://nathancahill.github.io/Split.js/

**Comment by ckrahe** [comment]
_OK so this is turning out to be our issue and not a Split.js issue.  I'll continue to document here for the curious._

After talking with more of our team, it turns out we experienced the same lag when we first started using 0.4.3.  It was indeed caused by our CSS2-style animation.  At the time, we solved it with a locally-modified Split.js, temporarily overriding our CSS during drag:

`var startDragging = function (e) {
  ...
  this.a.style.transition = 'none';
  this.b.style.transition = 'none';
  ...
},

var stopDragging = function () {
  ...
  this.a.style.transition = '';
  this.b.style.transition = '';
  ...
}`

We prefer to extend, not mod, so with 1.2.0 I'm now doing the temporary override in the available callbacks:

`Split(['#x', '#y'], {
  ...
  onDragStart: function() {
    document.querySelector('#x').style.transition = 'none';
    document.querySelector('#y').style.transition = 'none';
  },
  onDragEnd: function() {
    document.querySelector('#x').style.transition = '';
    document.querySelector('#y').style.transition = '';
  },
  ...
});`

Curiously, the lag remains. We've got more homework to do.

**Comment by rolandtoth** [comment]
It's the first time I tried Split.js and got the same issue. It turned out that there can be a "hover" or other event on the surrounding items (columns). So when I try to drag the gutter and the mouse gets outside of its area, these events kick in and stops the dragging event.

I've managed to get around this bug using a transparent pseudo :before that overlaps the entire page when the gutter is active. This way underlying elements can't interact at all, dragging is not stopped and everything is smooth.

```scss
.gutter {
  background-repeat: no-repeat;
  background-position: 50%;
  float: left;
  height: 100vh;
  position: relative;
  z-index: 300;
  &:active:before {
    content: "";
    position: fixed;
    left: 0;
    top: 0;
    bottom: 0;
    right: 0;
    background: transparent;
  }
}
```

Hope this helps.
